crates/core/src/client.rs

```diff
@@ -152,15 +167,31 @@ impl BeerusClient {
             let core_contract_addr = config.get_core_contract_address();
 
             loop {
-                let sn_root = sn_state_root_inner(&l1_client, core_contract_addr).await.unwrap();
-                let sn_block_num = sn_state_block_number_inner(&l1_client, core_contract_addr).await.unwrap();
+                let sn_root = wrapped_retry_sn_state_root_inner(
+                    &l1_client,
+                    core_contract_addr,
+                    config.timeout_secs,
+                    config.retry_limit,
+                    config.retry_secs,
+                )
+                .await
+                .unwrap();
```

Comment:
the main idea was to remove `unwrap()` and thus prevent the child thread from panicking

Reply:
Thanks @tonypony220 for your time! I was thinking of not avoiding `unwrap()` but avoiding silent panic in the thread which users are not aware of, as #561 illustrated. So `unwrap` still exists in the main loop, but in `wrapped_retry_*` functions it will catch errors and retry several times. When `retries` reaches the limit, it will return `Err` and make the thread panic and quit, and eventually make the main process quit since `panic = 'abort'` is added. What do you think, I can make the main loop completely ignore the errors, or retry indefinitely.

Reply:
i just refactored the `wrapped_retry` function so that the `unwrap()` can be resolved and removed. Now errors are handled in the `wrapped_retry` function, and it either returns the value or shuts the main process down. Moreover, parameters are set in `wrapped_retry` function so we can easily achieve indefinite retrying (namely, never panic).